src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitThis = sourceString.split(';');
+  const result = {};
+
+  for (const oneString of splitThis) {
+    let word = '';
+    let wordTwo = '';
+    let fazeTwo = false;
+
+    for (const symbol of oneString) {
+      if (symbol.toLowerCase() !== symbol.toUpperCase() || symbol === '-') {
+        if (fazeTwo !== true) {
+          word += symbol;
+        }
+      }
+
+      if (fazeTwo === true) {
+        wordTwo += symbol;
+      }
+
+      if (symbol === ':') {
+        fazeTwo = true;
+      }
```

Comment:
The logic to determine if a character should be part of the property or value is overly complex. Instead of checking the case of the character, you can split the string by ':' and then trim the resulting parts.